### PR TITLE
feat: add payment method and refund functions

### PR DESCRIPTION
**Pull request by Justintime50**
# Description

- Adds new billing functionality for ReferralCustomer users
  - `add_payment_method` can add a pre-existing Stripe bank account or credit card to your EasyPost account
  - `refund_by_amount` refunds your wallet by a dollar amount
  - `refund_by_payment_log` refunds you wallet by a PaymentLog ID

# Testing

Adds new tests that assert the proper errors since we won't have valid data for any of this.

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: easypost/referral.py [hunk 1]
```diff
@@ -106,6 +111,67 @@ def add_credit_card(
         )
         return convert_to_easypost_object(response)
 
+    @staticmethod
+    def add_payment_method(
```
Comment (jchen293) [inline easypost/referral.py:115]: These functions should be in the beta folder `referral`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
 # CHANGELOG
 
+## Next Release
+
+- Adds new billing functionality for ReferralCustomer users
```
Comment (jchen293) [inline CHANGELOG.md:5]:
```suggestion
- Adds new beta billing functionality for ReferralCustomer users
```

Nit: I would indicate this is `beta` in the changelog

**Review by jchen293 — DISMISSED** [review]
LGTM

File: CHANGELOG.md (hunk 2 above)
Comment (Justintime50) [reply CHANGELOG.md:5]: Will add that when we release

**Review by nwithan8 — DISMISSED** [review]

**Comment by Justintime50** [comment]
This is no longer valid since the functionality provided by Stripe is deprecated.

File: tests/test_referral.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
 REFERRAL_USER_PROD_API_KEY = os.getenv("REFERRAL_USER_PROD_API_KEY", "123")
```
Comment (nwithan8) [inline tests/test_referral.py:9]: Needs change to new env

File: tests/test_referral.py [hunk 4]
```diff
@@ -9,6 +9,7 @@
...
 
 
+# TODO: Rename this file to `test_referral_customer.py` and all test names need `customer` added to them
```
Comment (nwithan8) [inline tests/test_referral.py:12]: Why not do this now?

**Review by nwithan8 — CHANGES_REQUESTED** [review]

File: tests/test_referral.py (hunk 4 above)
Comment (Justintime50) [reply tests/test_referral.py:12]: So we can keep the test file matched up to the other file and change them both at the same time.

File: tests/test_referral.py (hunk 3 above)
Comment (Justintime50) [reply tests/test_referral.py]: Nice catch, only renamed half of them.

**Review by nwithan8 — APPROVED** [review]

**Review by jchen293 — DISMISSED** [review]

File: easypost/beta/referral.py [hunk 5]
```diff
@@ -109,6 +109,67 @@ def add_credit_card(
...
+        wrapped_params = {
+            "payment_method": {
+                "stripe_customer_id": stripe_customer_id,
+                "payment_method_reference": payment_method_reference,
+                "priority": priority,
+            }
+        }
+
+        response, api_key = requestor.request(
+            method=RequestMethod.POST,
```
Comment (nwithan8) [inline easypost/beta/referral.py:135]: This doesn't use the referral API key

**Review by nwithan8 — CHANGES_REQUESTED** [review]

File: easypost/beta/referral.py (hunk 5 above)
Comment (Justintime50) [reply easypost/beta/referral.py:135]: Per our discussion, this is setup correctly. You would have a specific `referral` client for this function call.

**Review by nwithan8 — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]

**Pull request merged**